### PR TITLE
[REF] web, project: add stacked line chart

### PR DESCRIPTION
**Pull request by fdardenne**
Added stacked option on line chart. 
The option already existed in Project for burndown graph, so I exported and adapted everything concerning stacked lines from Project to Web.

### CODE REVIEW

File: addons/project/static/src/burndown_chart/burndown_chart_view.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const viewRegistry = registry.category("views");
 
 const burndownChartGraphView = {
   ...graphView,
-  Renderer: BurndownChartRenderer,
+  Renderer: GraphRenderer,
```
Comment (SimonGenin) [inline addons/project/static/src/burndown_chart/burndown_chart_view.js:13]: not necessary

File: addons/web/static/src/views/graph/graph_renderer.js [hunk 2]
```diff
@@ -436,6 +443,9 @@ export class GraphRenderer extends Component {
                 suggestedMin: 0,
             },
         };
+        if (mode === "line") {
+            yAxe.stacked = stacked;
+        }
```
Comment (SimonGenin) [inline addons/web/static/src/views/graph/graph_renderer.js:448]: add as a boolean expr above

**Comment by SimonGenin** [comment]
@IT-Ideas for info, we're moving this feature to a generic one.

**Comment by IT-Ideas** [comment]
@SimonGenin, Nice 🙂
Thanks for the notification!

File: addons/web/static/src/views/graph/graph_renderer.js [hunk 3]
```diff
@@ -242,12 +242,13 @@ export class GraphRenderer extends Component {
...
      */
     getElementOptions() {
-        const { mode } = this.model.metaData;
+        const { mode, stacked } = this.model.metaData;
         const elementOptions = {};
         if (mode === "bar") {
             elementOptions.rectangle = { borderWidth: 1 };
         } else if (mode === "line") {
             elementOptions.line = { fill: false, tension: 0 };
+            elementOptions.line.fill = stacked;
```
Comment (SimonGenin) [inline addons/web/static/src/views/graph/graph_renderer.js:251]: `fill` is assigned to `false` then to `stacked`, you can make this a one liner.

File: addons/web/static/src/views/graph/graph_renderer.js [hunk 4]
```diff
@@ -351,6 +352,11 @@ export class GraphRenderer extends Component {
             dataset.fill = "origin";
             dataset.backgroundColor = hexToRGBA(getColor(0), 0.4);
         }
+        if (stacked) {
+            for (const dataset of data.datasets) {
+                dataset.backgroundColor = hexToRGBA(dataset.borderColor, 0.4);
+            }
+        }
```
Comment (SimonGenin) [inline addons/web/static/src/views/graph/graph_renderer.js:359]: While this is probably easier to read, it seems that you just loop over dataset a second time for "nothing".
Maybe check if you can put this in the for loop above, something like:
```js
if (stacked) {
    dataset.backgroundColor = hexToRGBA(dataset.borderColor, 0.4);
}
```
I ultimately agree that this code is somewhat gross.

File: addons/web/static/tests/views/graph_view_tests.js [hunk 5]
```diff
@@ -883,14 +887,140 @@ QUnit.module("Views", (hooks) => {
...
+        await click(stackButton);
+        assert.notOk(graph.model.metaData.stacked, "graph should be a classic line chart.");
+        assert.notOk(
+            getScaleY(graph).every((y) => y.stacked),
+            "The y axes should have stacked property set to false"
+        );
+    });
+
+    QUnit.test("Stacked line prop click false", async function (assert) {
+        assert.expect(4);
```
Comment (SimonGenin) [inline addons/web/static/tests/views/graph_view_tests.js:920]: not needed

File: addons/web/static/tests/views/graph_view_tests.js [hunk 6]
```diff
@@ -883,14 +887,140 @@ QUnit.module("Views", (hooks) => {
...
+            "backgroundColor",
+            "borderColor",
+            "originIndex",
+            "pointBackgroundColor",
+        ];
+        checkDatasets(assert, graph, keysToEvaluate, expectedDatasets);
+    });
+
+    QUnit.test("Stacked prop and default line chart", async function (assert) {
+        assert.expect(5);
```
Comment (SimonGenin) [inline addons/web/static/tests/views/graph_view_tests.js:969]: not needed (i think ? maybe the for loop at the end can cause problems?)

**Review by SimonGenin — CHANGES_REQUESTED** [review]

**Review by SimonGenin — APPROVED** [review]

**Pull request closed without merging**